src/rws_resource.cpp

```diff
@@ -51,4 +52,48 @@ std::ostream& operator<<(std::ostream& os, RAPIDResource const& resource)
         << ")";
 }
 
+std::ostream& operator<<(std::ostream& os, CFGDomain const& domain)
+{
+    return os << to_string(domain);
+}
+
+CFGDomain to_CFGDomain(std::string const& domain)
+{
+    static std::unordered_map<std::string, CFGDomain> const CFGDomainMap
+    {
+        {"EIO", D_EIO},
+        {"MMC", D_MMC},
+        {"MOC", D_MOC},
+        {"PROC", D_PROC},
+        {"SIO", D_SIO},
+        {"SYS", D_SYS},
+    };
+    auto it = CFGDomainMap.find(domain);
+    if (it != CFGDomainMap.end()) {
+        return it->second;
+    throw std::runtime_error("Unknown CFGDomain: " + domain);
+    }
```

Comment:
C++11 can do boundary checking automatically
```suggestion
    try
    {
        return CFGDomainMap.at(domain);
    } catch (std::out_of_range &ex)
    {
        throw std::runtime_error("Unknown CFGDomain: " + domain);
    }
```